refactor: improve split_shield hook making it more moddable

Darxo opened the pull request:
Extract shield damage calculation into new function. Hide tooltips for effects that are disabled.
Use this.getName() instead of a hard coded name for the combat log.

Commit:
refactor: improve split_shield hook making it more moddable

Extract shield damage calculation into new function.
Hide tooltips for effects that are disabled.
Use this.getName() instead of a hard coded name for the combat log.

LordMidas committed:
refactor: remove parameter from shield damage function and reorganize it

LordMidas commented on mod_reforged/hooks/skills/actives/split_shield.nut:
I pushed a commit that removes the `_user` parameter from the damage function, as it seemed unnecessary. I've also added handling for when it is the dummy player.

Darxo commented on mod_reforged/hooks/skills/actives/split_shield.nut:
```diff
@@ -3,13 +3,14 @@
 				type = "text",
 				icon = "ui/icons/shield_damage.png",
-				text = "Inflicts " + ::MSU.Text.colorGreen(this.getShieldDamage(this.getContainer().getActor())) + " damage to shields"
+				text = format("Inflicts%s damage to shields", ::MSU.isNull(this.getContainer()) || this.getContainer().getActor().getID() == ::MSU.getDummyPlayer().getID() ? "" : " " + ::MSU.Text.colorRed(shieldDamage))
```
You are missing a check for `::MSU.isNull(this.getContainer().getActor())` here before you access `getID`.
At least in the `RF_getShieldDamage` you added such a check.

LordMidas replied:
Vanilla sometimes seems to have that check and sometimes it doesn't. I personally think that check is unnecessary as an item_container should be expected to have an actor. I am not aware of a case where an is present in a container that doesn't have an actor, at least in the expected flow of the game. We can probably get away with removing that check from the other functions too, I guess I was just being "over safe" with it there.

Darxo replied:
I also like removing that check and seeing how it goes under the assumption that is really never happens.

LordMidas commented on mod_reforged/hooks/skills/actives/split_shield.nut:
```diff
@@ -108,6 +115,16 @@
+	q.RF_getShieldDamage <- function()
+	{
+		if (::MSU.isNull(this.getContainer()) || ::MSU.isNull(this.getContainer().getActor()))
+			return 1; // return a non-zero number so the tooltip gets added
```
I am not happy about my solution here to return 1. It will add the damage tooltip to the skill file always - but here we allow with null weapon returning 0 which will remove that entry.

So perhaps it is worth it to assume that this skill will always be attached to a weapon and therefore we can simply always access the weapon's damage and return that?

Darxo replied:
For the beast scenario we want something like `this.m.ShieldDamageOverride` which makes the `RF_getShieldDamage` function ignore anything else and return that value.

Apart from that I think we can expect a weapon to be present.

Darxo committed:
refactor: use msu function to retrieve random element from array

Darxo committed:
refactor: add _targetEntity argument for getShieldDamage function

This is currently not used by reforged.
This will improve submods who want different shield damage depending on
the target. For example by calling a 'buildPropertiesForUse'.

Darxo committed:
refactor: move back to simpler handling of shield damage tooltip

Change coloring to damage color.
Adjust id to 10

Darxo committed:
refactor: improve id counting of remaining entries

Darxo committed:
refactor: use msu's colorDamage function for fatigue damage

Darxo commented:
I pushed a few commits.
One of them rewrites your idea of displaying a default message when we are not sure if this skill applies shield damage.
